src/test/kotlin/org/rust/ide/intentions/MatchToIfLetIntentionTest.kt

```diff
@@ -0,0 +1,128 @@
+package org.rust.ide.intentions
+
+class MatchToIfLetIntentionTest : RustIntentionTestBase(MatchToIfLetIntention()) {
+    fun testUnavailable1() = doUnavailableTest(
+        """
+        enum MyOption {
+            Nothing,
+            Some(x),
+        }
+
+        fn main() {/*caret*/
+            let a = MyOption::Some(52);
+
+            match a {
+                MyOption::Some(x) => {},
+                Nothing => {}
+            }
+        }
+        """
+    )
+
+    fun testUnavailable2() = doUnavailableTest(
+        """
+        enum MyOption {
+            Nothing,
+            Some(x),
+        }
+
+        fn main() {/*caret*/
+            let a = MyOption::Some(52);
```

Comment:
I think the caret should be inside a match expr to really test the case of non-void arms.